docs-next/pages/apis/hooks.mdx

```diff
@@ -2,6 +2,455 @@ import { Markdown } from '../../components/Page';
+            afterChange: async args => { /* ... */ },
+
+            // Hooks for delete operations
+            validateDelete: async args => { /* ... */ },
+            beforeDelete: async args => { /* ... */ },
+            afterDelete: async args => { /* ... */ },
+          },
+        }),
+      },
+    }),
+  }),
+});
+```
+
+## Update and create hooks
+
+Update and create hooks are called for all `update` and `create` operations.
+When updating or creating multiple values the hooks are called individually for each item being updated or created.
+
+### resolveInput
+
+The `resolveInput` function is used to modify or augment the `data` values passed in to a `create` or `update` operation.
+
+This hook is the final stage in the [data resolving process](#resolved-data-stages), and is invoked after access control and field defaults are applied.
+
+For field hooks, the return value should be an updated value for that specific field.
+For list hooks, the return value should be a [`resolved data`](#resolved-data-stages) object.
+The result of `resolveInput` will be passed as `resolvedData` into the next stages of the operation.
+
+| Argument        | Description                                                                                                                                                                       |
+| :-------------- | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `listKey`       | The key of the list being operated on.                                                                                                                                            |
+| `fieldPath`     | The path of the field being operated on (field hooks only).                                                                                                                       |
+| `operation`     | The operation being performed (`'create'` or `'update'`).                                                                                                                         |
+| `originalInput` | The value of `data` passed into the mutation.                                                                                                                                     |
+| `existingItem`  | The currently stored item (`undefined` for `create` operations). This object is an unresolved list item. [list item API](./list-items) for more details on unresolved list items. |
+| `resolvedData`  | A [`resolved data`](#resolved-data-stages) object. The resolved data value after default values, relationship resolvers, and field resolvers have been applied.                   |
+| `context`       | The [`KeystoneContext`](./context) object of the originating GraphQL operation.                                                                                                   |
+
```

Comment:
General comment, I think we should explicitly define (or explain) the return type for each hook function as well as the arguments